src/comm.c

```diff
@@ -62,8 +63,13 @@ static void request_update() {
 static void in_received_handler(DictionaryIterator *received, void *context) {
   phone_contact = true;
   update_in_progress = false;
-  schedule_update(UPDATE_FREQUENCY);
-
+  Tuple* recency = dict_find(received, APP_KEY_RECENCY);
+  if (recency) {
+    int32_t next_update = UPDATE_FREQUENCY - recency->value->uint32 * 1000;
+    int32_t delay = next_update < 60000 ? 60000 : next_update;
+    schedule_update((uint32_t) delay);
+  } else
```

Comment:
Yep.  I'm embarrassed to have missed that.  It's been a joy looking at a Pebble app that looks like it's written by someone who is taking it seriously.  I'm so glad I don't have to do it myself.